assets/src/pages/room/VideochatSection.tsx

```diff
@@ -136,11 +153,15 @@ export const VideochatSection: FC<Props> = ({ peers, localPeer, showSimulcast, w
 
   const pinningApi = usePinning();
   const screenSharingStreams = prepareScreenSharingStreams(peers, localPeer);
-  useEffect(() => pinNewScreenShares(screenSharingStreams, pinningApi.pinIfNotAlreadyPinned));
 
   const allPeersConfig: MediaPlayerTileConfig[] = [localUser, ...mapRemotePeersToMediaPlayerConfig(peers)];
   const allTilesConfig: MediaPlayerTileConfig[] = allPeersConfig.concat(screenSharingStreams);
 
+  useEffect(() => {
```

Comment:
you could extract everything connected to pinning to separate file / hook. There is a lot of it in this file:
```
const takeOutPinnedTiles = (
  tiles: MediaPlayerTileConfig[],
  pinnedTileIds: string[]
): { pinnedTiles: MediaPlayerTileConfig[]; unpinnedTiles: MediaPlayerTileConfig[] } => {
  const { pinnedTiles, unpinnedTiles } = groupBy(tiles, ({ mediaPlayerId }) =>
    pinnedTileIds.includes(mediaPlayerId) ? "pinnedTiles" : "unpinnedTiles"
  );
  return { pinnedTiles: pinnedTiles ?? [], unpinnedTiles: unpinnedTiles ?? [] };
};

const pinNewScreenShares = (
  screenSharingStreams: ScreenShareTileConfig[],
  pinIfNotAlreadyPinned: (tileId: string) => void
) => {
  screenSharingStreams.map((tile) => tile.mediaPlayerId).forEach(pinIfNotAlreadyPinned);
};

const manageOneOnOnePinning = (
  mediaPlayerTiles: MediaPlayerTileConfig[],
  { pinnedTileIds, pinIfNotAlreadyPinned }: PinningApi
): void => {
  console.log("manageOneOnOnePinning");
  if (mediaPlayerTiles.length !== 2) return;

  const localUserTileId = mediaPlayerTiles.find((tile) => tile.streamSource === "local")?.mediaPlayerId;
  const remoteUserTileId = mediaPlayerTiles.find((tile) => tile.streamSource === "remote")?.mediaPlayerId;
  if (!localUserTileId || !remoteUserTileId) return;

  const isLocalPinned = pinnedTileIds.includes(localUserTileId);
  const isRemotePinned = pinnedTileIds.includes(remoteUserTileId);

  if (!isLocalPinned && !isRemotePinned) pinIfNotAlreadyPinned(remoteUserTileId); // pinning new peer
  // if (isLocalPinned && isRemotePinned) removePinnedEarlier(localUserTileId, remoteUserTileId); // switching the peer
};
...
const pinningApi = usePinning();

  useEffect(() => {
    pinNewScreenShares(screenSharingStreams, pinningApi.pinIfNotAlreadyPinned);
    manageOneOnOnePinning(allTilesConfig, pinningApi);
  }, [allTilesConfig, pinningApi, screenSharingStreams]);

  const { pinnedTiles, unpinnedTiles } = takeOutPinnedTiles(allTilesConfig, pinningApi.pinnedTileIds);
  const isAnyTilePinned = pinnedTiles.length > 0;
  const isAnyTileUnpinned = unpinnedTiles.length > 0;

  const shouldBlockPinning = allTilesConfig.length === 1;

  const betterPinning = useCallback(
    (tileId: string) => {
      if (allTilesConfig.length === 2 && pinningApi.pinnedTileIds[0]) {
        console.log(pinningApi.pinnedTileIds[0])
        pinningApi.unpin(pinningApi.pinnedTileIds[0]);
      }
      pinningApi.pin(tileId);
    },
    [allTilesConfig.length, pinningApi]
  );
```

Reply:
Done. Simplified some logic in `videochatSection.tsx` as well